Sources/Basics/FileSystem/AbsolutePath.swift

```diff
@@ -0,0 +1,330 @@
+    /// not interpret leading `~` characters as home directory specifiers).
+    /// The input string will be normalized if needed, as described in the
+    /// documentation for AbsolutePath.
+    public init(validating pathString: String) throws {
+        self.underlying = try .init(validating: pathString)
+    }
+
+    /// Initializes an AbsolutePath from a string that may be either absolute
+    /// or relative; if relative, `basePath` is used as the anchor; if absolute,
+    /// it is used as is, and in this case `basePath` is ignored.
+    public init(validating pathString: String, relativeTo basePath: AbsolutePath) throws {
+        self.underlying = try .init(validating: pathString, relativeTo: basePath.underlying)
+    }
+
+    /// Initializes the AbsolutePath by concatenating a relative path to an
+    /// existing absolute path, and renormalizing if necessary.
+    public init(_ absolutePath: AbsolutePath, _ relativeTo: RelativePath) {
+        self.underlying = .init(absolutePath.underlying, relativeTo.underlying)
+    }
+
+    /// Convenience initializer that appends a string to a relative path.
+    public init(_ absolutePath: AbsolutePath, validating relativePathString: String) throws {
+        try self.init(absolutePath, RelativePath(validating: relativePathString))
+    }
+
+    /// Directory component.  An absolute path always has a non-empty directory
+    /// component (the directory component of the root path is the root itself).
+    public var dirname: String {
+        self.underlying.dirname
+    }
+
+    /// Last path component (including the suffix, if any).  it is never empty.
+    public var basename: String {
+        self.underlying.basename
+    }
+
+    /// Returns the basename without the extension.
+    public var basenameWithoutExt: String {
+        self.underlying.basenameWithoutExt
+    }
+
+    /// Returns the basename dropping any possible  extension.
+    public func basenameWithoutAnyExtension() -> String {
+        var basename = self.basename
+        if let index = basename.firstIndex(of: ".") {
+            basename.removeSubrange(index ..< basename.endIndex)
+        }
+        return String(basename)
+    }
+
+    /// Suffix (including leading `.` character) if any.  Note that a basename
+    /// that starts with a `.` character is not considered a suffix, nor is a
+    /// trailing `.` character.
+    public var suffix: String? {
+        self.underlying.suffix
+    }
+
+    /// Extension of the give path's basename. This follow same rules as
+    /// suffix except that it doesn't include leading `.` character.
+    public var `extension`: String? {
+        self.underlying.extension
+    }
+
+    /// Unlike ``AbsolutePath//extension``, this property returns all characters after the first `.` character in a
+    /// filename. If no dot character is present in the filename or first dot is the last character, `nil` is returned.
+    public var allExtensions: [String]? {
```

Comment:
I guess this was a `Basics`-provided extension before? Might be worth keeping that separation?